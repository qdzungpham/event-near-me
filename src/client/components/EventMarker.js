import React from 'react';
import PropTypes from 'prop-types';
import { Marker, InfoWindow } from 'react-google-maps';

class EventMarker extends React.Component {
  state = {
    isOpen: false
  };

  onToggleOpen = () => {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  };

  onMouseOver = () => {
    const { event, onHoverMarker } = this.props;
    onHoverMarker(event.id);
  };

  onMouseOut = () => {
    const { onHoverMarker } = this.props;
    onHoverMarker('-1');
  };

  render() {
    const { event } = this.props;
    const { isOpen } = this.state;
    return (
      // eslint-disable-next-line
      <Marker
        id="marker"
        title={event.title}
        position={{ lat: event.coords[0], lng: event.coords[1] }}
        onClick={this.onToggleOpen}
        onMouseOver={this.onMouseOver}
        onMouseOut={this.onMouseOut}
      >
        {isOpen && (
          <InfoWindow id="infowindow" onCloseClick={this.onToggleOpen}>
            <h3>{event.title}</h3>
          </InfoWindow>
        )}
      </Marker>
    );
  }
}

EventMarker.propTypes = {
  event: PropTypes.object.isRequired,
  onHoverMarker: PropTypes.func.isRequired
};

export default EventMarker;
