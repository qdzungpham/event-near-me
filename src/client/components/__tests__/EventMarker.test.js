import React from 'react';
import { shallow } from 'enzyme';
import EventMarker from '../EventMarker';

describe('EventMarker', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <EventMarker
        event={{
          id: '2',
          title: 'title',
          dateTime: '2018-09-07T14:12:15.112Z',
          location: 'Brisbane',
          coords: [-24, 153],
          distance: 21,
          image: 'http://via.placeholder.com/100x100'
        }}
        onHoverMarker={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle isOpen when click marker', () => {
    const wrapper = shallow(
      <EventMarker
        event={{
          id: '2',
          title: 'title',
          dateTime: '2018-09-07T14:12:15.112Z',
          location: 'Brisbane',
          coords: [-24, 153],
          distance: 21,
          image: 'http://via.placeholder.com/100x100'
        }}
        onHoverMarker={jest.fn}
      />
    );

    wrapper.find('[id="marker"]').simulate('click');
    expect(wrapper.state()).toEqual({ isOpen: true });
  });
});
