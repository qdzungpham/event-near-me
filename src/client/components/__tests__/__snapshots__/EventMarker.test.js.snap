// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventMarker renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventMarker
    event={
      Object {
        "coords": Array [
          -24,
          153,
        ],
        "dateTime": "2018-09-07T14:12:15.112Z",
        "distance": 21,
        "id": "2",
        "image": "http://via.placeholder.com/100x100",
        "location": "Brisbane",
        "title": "title",
      }
    }
    onHoverMarker={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": false,
      "id": "marker",
      "onClick": [Function],
      "onMouseOut": [Function],
      "onMouseOver": [Function],
      "position": Object {
        "lat": -24,
        "lng": 153,
      },
      "title": "title",
    },
    "ref": null,
    "rendered": false,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": false,
        "id": "marker",
        "onClick": [Function],
        "onMouseOut": [Function],
        "onMouseOver": [Function],
        "position": Object {
          "lat": -24,
          "lng": 153,
        },
        "title": "title",
      },
      "ref": null,
      "rendered": false,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
